ui/src/app/subnets/views/SubnetDetails/UsedIPs/SubnetUsedIPs.tsx

```diff
@@ -1,7 +1,141 @@
+import { MainTable, Spinner } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+
 import TitledSection from "app/base/components/TitledSection";
+import type { RootState } from "app/store/root/types";
+import subnetSelectors from "app/store/subnet/selectors";
+import type {
+  Subnet,
+  SubnetDetails,
+  SubnetIP,
+  SubnetMeta,
+} from "app/store/subnet/types";
+import { IPAddressType, IPAddressTypeLabel } from "app/store/subnet/types/enum";
+import { NodeType, NodeTypeDisplay } from "app/store/types/node";
+
+export type Props = {
+  subnetId: Subnet[SubnetMeta.PK] | null;
+};
+
+export enum Labels {
+  IpAddresses = "IP addresses",
+  Type = "Type",
+  Node = "Node",
+  Interface = "interface",
+  Usage = "Usage",
+  Owner = "Owner",
+  LastSeen = "Last Seen",
+}
+
+const getType = (ip: SubnetIP) => {
+  return IPAddressTypeLabel[
+    IPAddressType[ip.alloc_type] as keyof typeof IPAddressTypeLabel
+  ];
+};
+
+const getUsage = (ip: SubnetIP) => {
+  const isContainer = ip.node_summary?.is_container;
+  const nodeType = ip.node_summary?.node_type;
+  if (nodeType === 1 && isContainer) {
+    return "Container";
+  }
+  if (nodeType) {
+    return NodeTypeDisplay[NodeType[nodeType] as keyof typeof NodeTypeDisplay];
+  } else {
+    return "Unknown";
+  }
+};
+
+const SubnetUsedIPs = ({ subnetId }: Props): JSX.Element => {
+  const subnet = useSelector((state: RootState) =>
+    subnetSelectors.getById(state, subnetId)
+  ) as SubnetDetails;
+  const loading = useSelector(subnetSelectors.loading);
+  const ip_addresses = subnet.ip_addresses;
 
-const SubnetUsedIPs = (): JSX.Element => (
-  <TitledSection title="Used IP addresses" />
-);
+  return (
+    <TitledSection title="Used IP addresses">
+      <MainTable
```

Comment:
Might be worth adding a `responsive` prop so that this is usable on smaller screens.

## before
![image](https://user-images.githubusercontent.com/7452681/153424953-b0002b48-3367-4814-ba8a-9995e6acf09b.png)


## after
![image](https://user-images.githubusercontent.com/7452681/153423272-a1b8925b-3c77-4aea-97bf-f344f7c5ac39.png)

Reply:
Done!